File: apps/api/app/schemas/payment_method.py
"""Payment method API schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class PaymentMethodResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    currency: str
    active: bool
    created_at: datetime = Field(serialization_alias="createdAt")

File: apps/web/src/types/payment-method.ts
/** Payment method as returned by the API (list and single). */
export interface PaymentMethod {
  id: string
  name: string
  currency: string
  active: boolean
  createdAt: string
}

/** List response: GET /api/v1/payment-methods */
export interface PaymentMethodListResponse {
  data: PaymentMethod[]
}
